### PR TITLE
[ET-1799] - Series pass panel added helper text link.

### PR DESCRIPTION
**Pull request by rafsuntaskin**
### 🎫 Ticket

[ET-1799] [ET-1788] 

### 🗒️ Description

* Added helper text label at the bottom of the Ticket list meta box panel.
* The tickets helper text is shown for all post types.
* The series pass helper text is shown only for Series post type.
* Updated snapshots.

### 🎥 Artifacts 
📷 screenshot(s):

Screenshot 2023-09-05 at 2 55 19 PM

Screenshot 2023-09-05 at 2 57 11 PM

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1799]: https://theeventscalendar.atlassian.net/browse/ET-1799?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ET-1788]: https://theeventscalendar.atlassian.net/browse/ET-1788?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/admin-views/editor/panel/list.php [hunk 1]
```diff
@@ -191,6 +191,18 @@ class="ticket_panel panel_base"
...
 	</div>
+	<div class="tec_ticket-panel__helper_text__wrap">
+		<p>
+			<?php
+			$helper_link = sprintf(
+				'<a href="%1$s" target="_blank" rel="noopener noreferrer ">%2$s</a>',
+				esc_url( 'https://evnt.is/manage-tickets' ),
+				esc_html__( 'Learn more about ticket management', 'event-tickets' )
+			);
+			echo sprintf( esc_html__( 'Create and manage Series Passes for this Series. %s', 'event-tickets' ), $helper_link );
```
Comment (lucatume) [inline src/admin-views/editor/panel/list.php:202]: This has to be filterable, probably an action, in which you would jump to customize the code for Series Passes from the FT project.

Comment (lucatume) [reply src/admin-views/editor/panel/list.php:202]: I propose we use the `tribe_events_tickets_after_new_ticket_panel` action here to render this.
For RSVP and Tickets (the default type) hook in the ET provider, for Series Passes, hook in the FT provider.

**Review by lucatume — CHANGES_REQUESTED** [review]

File: src/Tickets/Flexible_Tickets/Series_Passes.php [hunk 2]
```diff
@@ -854,4 +859,33 @@ public function filter_unticketed_count_query( string $query = null, string $pos
...
+	 *
+	 * @since TBD
+	 *
+	 * @param string $text The helper text with link.
+	 * @param WP_Post $post The Post object.
+	 *
+	 * @return string
+	 */
+	public function filter_tickets_panel_list_helper_text( string $text, WP_Post $post ): string {
+		if ( Series_Post_Type::POSTTYPE != $post->post_type ) {
```
Comment (lucatume) [inline src/Tickets/Flexible_Tickets/Series_Passes.php:874]: There is a `TEC\Tickets\Flexible_Tickets\Series_Passes\Meta` class: this method should be moved there similarly to [this one.](https://github.com/the-events-calendar/event-tickets/blob/a969603260b13418cb6271bb379b92fb56db21be/src/Tickets/Flexible_Tickets/Series_Passes.php#L728-L740)

File: src/Tribe/Metabox.php [hunk 3]
```diff
@@ -904,6 +906,43 @@ public static function localize_decimal_character() {
...
 	}
 
+	/**
+	 * Get the list panel helper text.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post $post The post object.
+	 *
+	 * @return string
```
Comment (lucatume) [inline src/Tribe/Metabox.php:916]:
```suggestion
	 * @return string The list panel  helper text.
```

File: src/Tribe/Metabox.php [hunk 4]
```diff
@@ -904,6 +906,43 @@ public static function localize_decimal_character() {
...
+	 */
+	public function get_list_panel_helper_text( $post ) {
```
Comment (lucatume) [inline src/Tribe/Metabox.php:918]: Type-hint whenever possible.

```suggestion
	public function get_list_panel_helper_text( WP_Post $post ) : string {
```

Comment (lucatume) [reply src/Tribe/Metabox.php:918]: Why is this method  `public`? It looks like it should be `protected`.

File: src/admin-views/editor/panel/list.php [hunk 5]
```diff
@@ -191,6 +193,20 @@ class="ticket_panel panel_base"
 			<?php endif; ?>
 		</div>
 	</div>
+	<div class="tec_ticket-panel__helper_text__wrap">
+		<p>
+			<?php
+			echo wp_kses( $helper_text, [
```
Comment (lucatume) [inline src/admin-views/editor/panel/list.php:199]: Too much indentation?

File: src/admin-views/editor/panel/ticket.php [hunk 6]
```diff
@@ -26,7 +26,7 @@
  * @var string                             $ticket_type                      The type of Ticket the form is for.
  */
 
-$ticket_type = $ticket_type ?: 'default';
+$ticket_type = $ticket_type ?? 'default';
```
Comment (lucatume) [inline src/admin-views/editor/panel/ticket.php:29]: Why this change?
`?:` means "if the ticket_type is empty use default", while `??` means "if the ticket_type is NULL use default". We want to use "default" when the ticket type is an empty string.

**Review by lucatume — CHANGES_REQUESTED** [review]
Some notes, some questions.

File: src/Tribe/Metabox.php (hunk 4 above)
Comment (rafsuntaskin) [reply src/Tribe/Metabox.php:918]: As we are returning a filtered value here, therefore, I have skipped the type hinting for the return type in the signature. 

Should we always do type hinting with filters as well?

File: src/admin-views/editor/panel/ticket.php (hunk 6 above)
Comment (rafsuntaskin) [reply src/admin-views/editor/panel/ticket.php:29]: When the template is loaded for New Events/posts, there are no tickets associated with that and we don't pass any:
https://github.com/the-events-calendar/event-tickets/blob/0b7bb6078d32708fa53ec8d53cd0c790b1a6c1f0/src/Tribe/Metabox.php#L109

We can either pass a `ticket_type` var on the linked location or check for `null` to avoid the warning on the new post/event pages.

**Review by lucatume — APPROVED** [review]
